google/cloud/storage/examples/storage_bucket_samples.cc

```diff
@@ -927,7 +927,12 @@ void LockRetentionPolicy(google::cloud::storage::Client client, int& argc,
                 << ", status=" << original.status() << std::endl;
       return;
     }
-    client.LockBucketRetentionPolicy(bucket_name, original->metageneration());
+    StatusOr<void> status = client.LockBucketRetentionPolicy(
```

Comment:
This operation returns a bucket resource. I'm following up on the API reference docs. I missed this during the first review.. It would be one less StatusOr<void>.

Reply:
Well, it should be `StatusOr<BucketMetadata>` in that case.  But the documentation says the response is empty:

https://cloud.google.com/storage/docs/json_api/v1/buckets/lockRetentionPolicy

Which documentation are you looking at?  In any case, changing the return type to `StatusOr<BucketMetadata>` should be a separate bug and PR, we need to change a lot more than just whatever is here in that case.

Reply:
Same documentation. I've sent a CL to update the reference documentation.

Understood. I'll file an issue and approve this PR.